src/util/hw_topo/mpir_hw_topo.c

```diff
@@ -0,0 +1,948 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
```

Comment:
It is just a single file, so let's flatten it to `src/util/mpir_hw_topo.c`.

Reply:
That directory also contains `mpir_netloc.c`

Reply:
let's move both